base/reflection.jl

```diff
@@ -31,6 +33,30 @@ Base.LinAlg
 """
 module_parent(m::Module) = ccall(:jl_module_parent, Ref{Module}, (Any,), m)
 
+"""
+    module_root(m::Module) -> Module
+
+Get the root module enclosing `m`: this is the top-level module loaded by
+`import` or `using` which ultimately create `m`. This root module is found by
+iteratively taking a module's parent until self-parented module is found.
```

Comment:
I didn't mean to leave this in the PR – it was something I wrote and then didn't end up using, but it's somewhat useful, so maybe we should leave it in.